src/parquet/column/scanner.h

```diff
@@ -45,7 +45,7 @@ class Scanner {
       values_buffered_(0),
       reader_(reader) {
     // TODO: don't allocate for required fields
-    def_levels_.resize(reader->descr()->is_optional() ? batch_size_ : 0);
+    def_levels_.resize(!reader->descr()->is_required() ? batch_size_ : 0);
```

Comment:
If the `ColumnDescriptor` is REQUIRED inside a nested schema tree, it will have definition levels